stdlib/list.mli

```diff
@@ -528,8 +528,10 @@ val fast_sort : ('a -> 'a -> int) -> 'a list -> 'a list
  *)
 
 val sort_uniq : ('a -> 'a -> int) -> 'a list -> 'a list
-(** Same as {!sort}, but also remove duplicates.
+(** Same as {!sort}, but also remove duplicates (in case of multiple
+    elements comparing equal, keep only the first).
     @since 4.02 (4.03 in ListLabels)
+    @before 5.4 the kept duplicates were not guaranteed to be the first.
  *)
```

Comment:
A proposal:

```
(** Same as {!sort}, but also removes duplicates: if multiple
   elements are comparing equal, only the first is kept in the
   sorted list.

    @since 4.02 (4.03 in ListLabels)

    @before 5.4 the element kept was not necessarily the first
 *)
```

Whatever you choose, please update the ListLabels documentation as well, the two docs should be identical.

Reply:
Done in both mli's (with almost your proposal).

I resolved conflicts by rebasing trunk, I hope it's ok with the conventions of the house. You may want to squash-and-merge the multiple commit mess.